### PR TITLE
Create tree.num_children() method

### LINKED ISSUE #431: Tree.num_children(node), or Tree.is_unary(node) function?
With tree sequences now allowing unary nodes, I find that I'm relatively regularly checking if a node is unary. I might be wrong, but it seems to me that it might be a little inefficient to keep doing `len(tree.children(u))` for this. If so, would it be sensible to simply report the number of children somehow, e.g. via a new method `Tree.num_children(u)`? Or (presumably even more efficiently, but much less generally), `Tree.is_unary(u)`?

### PR DESCRIPTION
**Pull request by hyanwong**
Fixes #431

### CODE REVIEW

**Comment by hyanwong** [comment]
Sorry - forgot to test. Will correct now.

File: python/CHANGELOG.rst [hunk 1]
```diff
@@ -13,6 +13,8 @@ In development
   to TypeError, or ValueError if the missing data character clashes
   (:user:`hyanwong`, :pr:`426`).
 
+- Convenient fast access to the number of children of a node in a tree with a
```
Comment (jeromekelleher) [inline python/CHANGELOG.rst]: Too much detail here: user's don't care how it's implemented.

File: python/_tskitmodule.c [hunk 2]
```diff
@@ -8136,6 +8136,26 @@ Tree_get_mrca(Tree *self, PyObject *args)
     return ret;
 }
 
+static PyObject *
+Tree_get_num_children(Tree *self, PyObject *args)
+{
+    PyObject *ret = NULL;
+    size_t num_children;
```
Comment (jeromekelleher) [inline python/_tskitmodule.c]: could make this an unsigned int here and avoid the cast below.

File: python/tests/test_highlevel.py [hunk 3]
```diff
@@ -1579,6 +1579,11 @@ def test_func(t, u):
                 self.assertEqual(l1, l2)
                 self.assertEqual(t.get_num_samples(u), len(l1))
 
+    def test_num_children(self):
+        for tree in self.get_tree():
+            for u in tree.nodes():
+                assert tree.num_children(u) == len(tree.children(u))
```
Comment (jeromekelleher) [inline python/tests/test_highlevel.py]: self.assertEqual

File: python/_tskitmodule.c (hunk 2 above)
Comment (hyanwong) [reply python/_tskitmodule.c]: Good catch - copy-paste snafu. Done

File: python/CHANGELOG.rst (hunk 1 above)
Comment (hyanwong) [reply python/CHANGELOG.rst]: Shortened

**Pull request merged**